File: mcdowell/python/ch11-06.py
def findElement(matrix, target):

    row = 0
    col = len(matrix[0]) - 1


    while row < len(matrix)  and col >= 0:
        if matrix[row][col] == target:
            return row, col
        elif matrix[row][col] > target:
            col -= 1
        else:
            row += 1

    return False




print findElement([
    [1,2,3,4],
    [11,12,13,14],
    [14,15,23,24],
], 15)

File: mcdowell/php/ch11-06.php
<?php

function findElement($matrix, $target) {
    $row = 0;
    $col = count($matrix[0]) - 1;

    while ($row < count($matrix) && $col >= 0) {
        if ($matrix[$row][$col] == $target) {
            return [$row, $col];
        }
        else if ($matrix[$row][$col] > $target) {
            $col --;
        }
        else {
            $row ++;
        }
    }
    return false;
}




$inputs = [
    [1,2,3,4,5],
    [5,6,7,8,9],
    [6,7,8,9,10],
    [11,20,23,34,55]
];


print_r(findElement($inputs, 20));